static/js/components/Navbar.js

```diff
@@ -37,6 +38,36 @@ export default class Navbar extends React.Component {
     return empty === true ? undefined : <UserMenu />;
   };
 
+  renderMMHeaderLogo: Function = (link: string, pathname: string): React$Element<*> => {
+    if (!PROFILE_PAGE_REGEX.test(pathname)) {
+      return (
+        <Link to={link}>
+          <img src="/static/images/mit-logo-transparent.svg" alt="MIT" />
+        </Link>
+      );
+    }
+
+    return  <img src="/static/images/mit-logo-transparent.svg" alt="MIT" />;
+  }
+
+  renderMMHeaderText: Function = (link: string, pathname: string): React$Element<*> => {
+    if (!PROFILE_PAGE_REGEX.test(pathname)) {
+      return (
+        <span className="mdl-layout-title">
+          <Link to={link}>
+            MicroMasters
+          </Link>
+        </span>
+      );
+    }
+
+    return (
+      <span className="mdl-layout-title">
+        MicroMasters
+      </span>
+    );
+  }
```

Comment:
Why not have the header text and the image each be rendered by the same method, with a different method for each case?

Then you'd have something like:

```js
renderProfileHeader = () => ([
  <span />,
  <img />
]);

renderNormalHeader = () => ([
  <span />,
  <img />
]);
```

then in the JSX for the `render` method you'd just have something like:

```js
{ PROFILE_REGEX.test(pathname) ? renderProfileHeader() : renderNormalHeader() }
``` 

What do you think?
 
I like this kind of approach because the methods are simpler and therefore easier to refactor.